subsys/canbus/canopen/CO_driver.h

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) 2019 Vestas Wind Systems A/S
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef CO_DRIVER_TARGET_H
+#define CO_DRIVER_TARGET_H
+
+/*
+ * Zephyr RTOS CAN driver interface and configuration for CANopenNode
+ * CANopen protocol stack.
+ *
+ * See CANopenNode/stack/drvTemplate/CO_driver.h for API description.
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <zephyr.h>
+#include <zephyr/types.h>
+#include <device.h>
+#include <toolchain.h>
+
+/* Use static variables instead of calloc() */
+#define CO_USE_GLOBALS
+
+/* Use Zephyr provided crc16 implementation */
+#define CO_USE_OWN_CRC16
+
+/* Use SDO buffer size from Kconfig */
+#define CO_SDO_BUFFER_SIZE CONFIG_CANOPEN_SDO_BUFFER_SIZE
+
+/* Use trace buffer size from Kconfig */
+#define CO_TRACE_BUFFER_SIZE_FIXED CONFIG_CANOPEN_TRACE_BUFFER_SIZE
+
+#ifdef CONFIG_CANOPEN_LEDS
+#define CO_USE_LEDS 1
+#endif
+
+#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+#define CO_LITTLE_ENDIAN
+#elif __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
+#define CO_BIG_ENDIAN
+#else
+#error "Unsupported endianness"
+#endif
+
+typedef bool          bool_t;
+typedef float         float32_t;
+typedef long double   float64_t;
+typedef char          char_t;
+typedef unsigned char oChar_t;
+typedef unsigned char domain_t;
+
+typedef struct {
+	u16_t ident;
+	u8_t DLC;
+	u8_t data[8];
+} CO_CANrxMsg_t;
```

Comment:
We could probably use typedef struct zcan_frame CO_CANrxMsg_t;
(only a remark)

Reply:
No, unfortunately we cannot. The CANopenNode stack expects `struct CO_CANrxMsg_t` to contain at least `data` and `DLC` members (`struct zcan_frame` has `data` and `dlc`).